frontend/services/api.ts

```diff
@@ -21,12 +21,16 @@ const API = axios.create({
 
 const onResponseSuccess = (response) => {
   try {
-    const parsedData = JSON.parse(response);
     return {
-      data: dataFormatter.deserialize(parsedData),
-      meta: parsedData.meta,
+      data: {
+        // Deserialized data
+        data: !!response.data && dataFormatter.deserialize(response.data),
```

Comment:
`data`'s type would be `boolean | whateverDeserializeReturns`. Shouldn't it be something like this?
```suggestion
        data: response.data ? dataFormatter.deserialize(response.data) : null,
```